pkg/poll/poll.go

```diff
@@ -27,37 +28,55 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-const (
-	RetryInterval = time.Second * 5
-	Timeout       = time.Second * 30
-)
+//go:generate mockgen -source=poll.go -package=poll -destination=mock_poll_api.go
+
+type PollActions interface {
+	ForResourceUnavailability(obj *unstructured.Unstructured) error
+	ForResource(obj *unstructured.Unstructured) error
```

Comment:
Wondering if we should keep hiding most polling functions behind `ForResource`... But I'm not sure this fits the scope of this PR.

Reply:
currently other packages are using the ForResource API. So for this PR i am keeping it as is, but we may later change it , once we start dealing with other packages